### ISSUE TITLE
useForm helper on 422 validation errors in the helper won't update

### ISSUE THREAD

**Issue by stephenkhoo** [inertiajs/inertia#1521]
### Version:

- `@inertiajs/svelte` version: 1.0.2, 1.0.3

### Describe the problem:

When a form is submitted, errors from backend passed to the helper error bag with setError
However, when we fill in a few fields, and resubmit, the error supposedly clear and set with new errors.
But it didn't updated.

### Steps to reproduce:

useForm to prepare data and submit to a validation protected request
Submit without data, and let the $form.errors shows up.
Add in a few data (supposed to remove some of the errors), and submit again
The errors remain.

### What we've tried

We tried to use options.onError
To set the error again

`$form.clearErrors().setError(errors)`

The problem remains

Then we attempt to break the chain

```
$form.clearErrors()
$form.setError(errors)
```

Then it works.
Not sure if it is cause by some bug within svelte when chain in this way.
However, when we do not chain it, it will work as expected

**Comment by weavdale** [comment]
I ran into this problem also. I thinking about submitting a PR for the fix.

The problem is the useForm helper is improperly using the Svelte store. On line 75 the `clearErrors` function is calling the `setStore` helper function which then calls the `update` function of the Svelte store. This is the correct way to update a value in the Svelte store. The store will then notify all subscribers of the change.

The problem comes on line 68 of the `setError` function. It is directly referencing `this.errors` variable to get a list of the current errors to add back into the store along with any new errors. This is not the correct way to get current values from the Svelte store. You need to subscribe to the store or use the store's `update` function which passes in the stores current values to the callback function. In this case when you call `clearErrors` and then immediately `setError` the internal variable `this.errors` hasn't been updated yet and still contains the old errors that were cleared in the `clearErrors` function.

There is more than one way to fix this, but my recommendation is to eliminate the helper function `setStore` to update the store and to explicitly call the store's `update` function every time you want to update the store. This way you always have access to the store's current values.

The new `setError` and `clearErrors` would look like this. Note, all other functions using `setStore` would also need updated.

```
function setError(key, value) {
  store.update((currentState) => ({
    ...currentState,
    errors: Object.assign(
      {},
      currentState.errors,
      value ? { [key]: value } : key
    ),
  }));

  return this;
}

function clearErrors(...fields) {
  store.update((currentState) => ({
    ...currentState,
    errors: Object.keys(currentState.errors).reduce(
      (carry, field) => ({
        ...carry,
        ...(fields.length > 0 && !fields.includes(field)
          ? { [field]: currentState.errors[field] }
          : {}),
      }),
      {}
    ),
  }));

  return this;
}
```

**Comment by stephenkhoo** [comment]
I think it would be a good to have the PR.
As this is actually a hidden bug.

**Comment by pedroborges** [comment]
Already fixed by #1610 ✅
